### PR TITLE
Fix for Tensorflow 2.1

### PR DESCRIPTION
**Pull request by Zahlii**
See also https://github.com/slundberg/shap/issues/1055

Using this version, I was able to successfully calculate SHAP values for a complicated NN based on TF 2.1.0

### CODE REVIEW

File: shap/explainers/deep/deep_tf.py [hunk 1]
```diff
@@ -146,8 +151,7 @@ def __init__(self, model, data, session=None, learning_phase_flags=None):
             else:
                 self.expected_value = tf.reduce_mean(self.model(self.data), 0)
 
-        if not tf.executing_eagerly():
-            self._init_between_tensors(self.model_output.op, self.model_inputs)
+        self._init_between_tensors(self.model_output.op, self.model_inputs)
```
Comment (slundberg) [inline shap/explainers/deep/deep_tf.py]: Did you try this in eager mode?

**Comment by Zahlii** [comment]
Hi, while running our internal test suite I still noticed some problems related to shap, will try to fix them and update this MR accordingly.

File: shap/explainers/deep/deep_tf.py (hunk 1 above)
Comment (FRUEHNI1) [reply shap/explainers/deep/deep_tf.py]: Yes - thats why I included this change. If I do not run this code, I get the following error downstream: 
image

**Comment by Zahlii** [comment]
Closing since its now obsolete, will follow-up with a newer PR.

**Pull request closed without merging**